pkg/db/interface.go

```diff
@@ -46,10 +46,19 @@ type VizierDBInterface interface {
 	DBInit()
 	SelectOne() error
 
-	GetStudyConfig(string) (*api.StudyConfig, error)
+	GetStudyMetrics(string) ([]string, error)
+	GetStudyIDsTypesList() ([]string, []string, error)
+	GetStudy(string) (*api.StudyConfig, error)
 	GetStudyList() ([]string, error)
-	CreateStudy(*api.StudyConfig) (string, error)
-	UpdateStudy(string, *api.StudyConfig) error
+
+	GetHPStudyConfig(string) (*api.StudyConfig, error)
+	CreateHPStudy(*api.StudyConfig) (string, error)
+	UpdateHPStudy(string, *api.StudyConfig) error
+
+	/* APIs for NAS */
+	GetNASStudyConfig(string) (*api.StudyConfig, error)
+	CreateNASStudy(*api.StudyConfig) (string, error)
+	UpdateNASStudy(string, *api.StudyConfig) error
```

Comment:
I think we should drop NAS and HP interface here.  It is enough to expose only study for VizierDBInterface.
Saying: only CreateStudy is enough, we can drop CreateNASStudy and CreateHPStudy in VizierDBInterface and the two can be subfunctions of CreateStudy. The difference between them is that: which field should processed before insert. We can even process in a unified way in CreateStudy to handle both case:
```
var configs, nasConfig string
if in.ParameterConfigs {
  configs, err = (&jsonpb.Marshaler{}).MarshalToString(in.ParameterConfigs)
}
if in.NasConfig {
  nasConfig, err = (&jsonpb.Marshaler{}).MarshalToString(in.NasConfig)
}
```

Reply:
I agree with @hougangliu. For example, in  CreateNASStudy and CreateHPStudy, most of codes are duplicate. We can merge them.

Reply:
Great catch! Didn't notice it firstly